File: Probleme curs/Probleme curs/Kotlin/M. Pod.kt
//Model pod
interface DrawAPI{
    fun drawCircle(radius: Int,x:Int,y:Int)
}
abstract class Shapes(protected val drawAPI:DrawAPI){
    abstract fun draw()
}
class Circles(val x: Int,val y:Int, val radius:Int,drawAPI:DrawAPI):Shapes(drawAPI){
    override fun draw(){
        drawAPI.drawCircle(radius,x,y)
    }
}
class GreenCircle:DrawAPI{
    override fun drawCircle(radius: Int,x:Int,y:Int){
        println("Drawing Circle[color:green,radius: $radius,x: $x,y: $y")
    }
}
class RedCircle:DrawAPI{
    override fun drawCircle(radius: Int,x:Int,y:Int){
        println("Drawing Circle[color:red,radius: $radius,x: $x,y: $y")
    }
}
fun main(args: Array<String>){
    val redCircle=Circles(100,100,10,RedCircle())
    val greenCircle=Circles(100,100,10,GreenCircle())
    redCircle.draw()
    greenCircle.draw()
}

File: Probleme curs/Probleme curs/Python/M. Pod.py
#Modelul Pod
class DrawingAPIOne:
    def drawCircle(self,x,y,radius):
        print("API 1 deseneaza un cerc la({},{}) cu raza {}".format(x,y,radius))
class DrawingAPITwo:
    def drawCircle(self,x,y,radius):
        print("API 2 deseneaza un cerc la ({},{}) cu raza {}".format(x,y,radius))
class Circle:
    def __init__(self,x,y,radius,drawingAPI):
        self._x=x
        self._y=y
        self._radius=radius
        self._drawingAPI=drawingAPI
    def draw(self):
        self._drawingAPI.drawCircle(self._x,self._y,self._radius)
    def scale(self,percent):
        self._radius*=percent

circle1=Circle(0,0,2,DrawingAPIOne())
circle1.draw()
circle2=Circle(1,3,3,DrawingAPITwo())
circle2.draw()